javascript/ql/src/semmle/javascript/dataflow/internal/AccessPaths.qll

```diff
@@ -39,12 +39,23 @@ private PropertyName getPropertyName(PropAccess pacc) {
   )
 }
 
+private SsaVariable getRefinedVariable(SsaVariable variable) {
```

Comment:
Perhaps this would read more naturally as a binary predicate? Then you could use `not refines(var, _)` in `MkSsaRoot` below.

Reply:
I think I'll pass on this. I find `getARefinementOf*(var).getAUseIn(bb)` to be quite readable.

Reply:
OK, but note that that's not the predicate I'm talking about.

Reply:
Ah, I thought you wanted to merge the two predicates. Still think predicates with results are so much easier to work with, though.

Reply:
Like I said, up to you.